src/primitive/measure.jl

```diff
@@ -23,6 +23,23 @@ mutable struct Measure{N, K, OT, RNG} <: PrimitiveBlock{N}
     end
 end
 
+"""
+    choperator(m::Measure, op::AbstractBlock)
+
+change the measuring `operator`. It will also discard existing measuring results.
+"""
+function choperator(m::Measure{N}, op::AbstractBlock) where N
+    Measure(N; rng=m.rng, operator=op, locs=m.locations, collapseto=m.collapseto, remove=m.remove)
```

Comment:
add an empty fallback so this can be used in tree traversing functions?

```jl
choperator(x::AbstractBlock, op::AbstractBlock) = x
```

Reply:
This is only for `Measure`, it means measure some operator. For example, we want to measure on different bases in some cases.